### DISCUSSION TITLE
What causes `ERROR ironbar::bar: `

### DISCUSSION THREAD

**Discussion by JustSimplyKyle** [JakeStanger/ironbar#73] [Q&A]
I have a script, and the error happens when I left click on it
`2023-02-11T05:20:56.642728Z ERROR ironbar::bar: 
2023-02-11T05:20:56.982159Z ERROR ironbar::bar:`
```corn
    $expand_sound = {
        type = "script"
        mode = "poll"
        interval = 100
        cmd = "zsh -c 'if /home/kyle/.config/ironbar/states/soundbar/soundbar_state ; then; echo '';else;echo '';fi'"
        on_click_left = "/home/kyle/.config/ironbar/states/soundbar/expand_soundbar"
    }
        $cava = {
        type = "script"
        mode = "watch"
        cmd = "/home/kyle/.config/eww/scripts/cava"
        show_if = $check_sound_state
    }
```

the expand_soundbar is just a simple lock script
```bash
LOCK="/tmp/ironbar_soundbar.lock"

if [ -f "$LOCK" ]; then
	rm -rf "$LOCK"
	exit 0
else
	touch "$LOCK"
	exit 1
fi
```
soundbar_state is even simpler
```bash
LOCK="/tmp/ironbar_soundbar.lock"
if [ -f "$LOCK" ]; then
	exit 0
else
	exit 1
fi
```

**Comment by JakeStanger** [comment]
I *suspect* it is because you are exiting `1` when creating the lock. Since this is a non-zero exit code, you're basically declaring the script failed to run. Ironbar picks this up as an error, and prints the `stderr`  from your script. This shows blank because nothing gets written to it.

**Reply by JustSimplyKyle** [reply]
But how would that explain when this shows up when it should be exiting 0?

[Kooha-2023-02-11-22-10-03.webm](https://user-images.githubusercontent.com/68589851/218262500-e58018b0-f6c7-4945-b77b-a4abeb8e0a89.webm)

**Reply by JakeStanger** [reply]
Are you sure? I've just tested this myself and it doesn't. If you change the `exit 1` to an `exit 0` in `expand_soundbar` the problem goes away, for me at least.
